File: Iniciante/1118.py
novo = 1
nota_valida = 0
media = 0
while novo == 1:
    while nota_valida != 2:
        nota = float(input())
        if nota < 0 or nota > 10:
            print('nota invalida')
        else:
            media += nota / 2
            nota_valida += 1
    novo = int(input('novo calculo (1-sim 2-nao)'))
print('media = {:.2f}'.format(media)) 

File: 1118.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main(){
    int contadorPositivas, continuar;
    double notas, soma;
    while (continuar != 2){
        cin >> notas;
        if (notas >= 0 && notas <= 10){
            soma += notas;
            contadorPositivas++;
        } else {
            cout << "nota invalida" << endl;
        } if (contadorPositivas == 2){
            cout << fixed << setprecision(2);
            cout << "media = " << soma / 2<< endl;
            cout << "novo calculo (1-sim 2-nao)" << endl;
            cin >> continuar;
            while (continuar < 1 || continuar > 2){
                cout << "novo calculo (1-sim 2-nao)" << endl;
                cin >> continuar;
            }
            if (continuar == 1){
                soma = 0;
                contadorPositivas = 0;
            }
        }
    }
    return 0;
}
